### PR TITLE
Improve RSS feed

### LINKED ISSUE #1870: Improvement to RSS feed
Hello,

Very interesting project. I'd like to suggest that the RSS feed could be improved in a few ways.

For each item, I'd like to see included:
- description
- a link to each item's writeup in the database
- item date 

The following sample is from https://incidentdatabase.ai/rss.xml (last built Thu, 13 Apr 2023 18:10:39 GMT)

```xml
<item>
  <title><![CDATA[Italian privacy regulator bans ChatGPT]]></title>
  <link>https://www.politico.eu/article/italian-privacy-regulator-bans-chatgpt/</link>
  <guid isPermaLink="false">d8463cdc-5971-51c5-997d-192c0c46063b</guid>
  <enclosure url="https://res.cloudinary.com/pai/image/upload/d_fallback.jpg,w_800/v1/reports/www.politico.eu/cdn-cgi/image/width%3D1200%2Cheight%3D630%2Cfit%3Dcrop%2Cquality%3D80%2Conerror%3Dredirect/wp-content/uploads/2023/03/31/iStock-1468873339-scaled.jpg" length="0" type="image/jpeg"/>
</item>
<item>
  <title><![CDATA[We tested a new ChatGPT-detector for teachers. It flagged an innocent student.]]></title>
  <link>https://www.washingtonpost.com/technology/2023/04/01/chatgpt-cheating-detection-turnitin/</link>
  <guid isPermaLink="false">156f90c9-0957-53c3-989b-eb5368ff3f11</guid>
  <enclosure url="https://res.cloudinary.com/pai/image/upload/d_fallback.jpg,w_800/v1/reports/upload.wikimedia.org/wikipedia/commons/thumb/9/93/The_Logo_of_The_Washington_Post_Newspaper.svg/500px-The_Logo_of_The_Washington_Post_Newspaper.svg.png" length="0" type="image/jpeg"/>
</item>
```

## Description

The example feed (https://deploy-preview-97--aiid.netlify.app/rss.xml), from [#45](https://github.com/responsible-ai-collaborative/aiid/issues/45#issuecomment-786763554) demonstrated including a description in the `<description>` tag, however the current implementation (based on #97) doesn't include description.

## Link

Right now there is no way to go from an item in the RSS feed to the Incident Database. It would be nice if the abbreviated description included a link to the IncidentDatabase page about that report.

Currently the RSS feed includes a link to the "source". I think this is a good idea and worth keeping.

E.g. for "Italian privacy regulator bans ChatGPT", the link is to https://www.politico.eu/article/italian-privacy-regulator-bans-chatgpt/. I think it would be good to also include a link the particular report (e.g. https://incidentdatabase.ai/cite/513/#r2900) and/or to the incident (e.g. Incident 513: ChatGPT Banned by Italian Authority Due to OpenAI's Lack of Legal Basis for Data Collection and Age Verification / https://incidentdatabase.ai/cite/513

## Date

Currently RSS readers have no reference about item dates, other than when their reader first saw them. Including the report date would improve this.

### PR DESCRIPTION
**Pull request by clari182**
- closes #1870 

- Added description (shorten) and link to incident inside the description tag
- Added pubDate tag (incident's date submitted)

https://deploy-preview-2050--staging-aiid.netlify.app/rss.xml

### CODE REVIEW

**Review by cesarvarela — COMMENTED** [review]
Can you add a test for this? I I think we have none.

**Comment by clari182** [comment]
@cesarvarela done! Test added

File: site/gatsby-site/cypress/e2e/integration/pages.cy.js [hunk 1]
```diff
@@ -148,4 +148,39 @@ describe('Pages', () => {
...
+  it('Should generate a valid RSS feed', () => {
+    cy.request('/rss.xml').then((response) => {
+      expect(response.status).to.eq(200);
+      expect(response.headers['content-type']).to.include('application/xml');
+
+      const xml = response.body;
+
+      const parsedXml = Cypress.$.parseXML(xml);
+
+      console.log(parsedXml);
```
Comment (cesarvarela) [inline site/gatsby-site/cypress/e2e/integration/pages.cy.js:161]: Please delete this.

**Review by cesarvarela — APPROVED** [review]
One nit and it gets merged.

File: site/gatsby-site/cypress/e2e/integration/pages.cy.js (hunk 1 above)
Comment (cesarvarela) [reply site/gatsby-site/cypress/e2e/integration/pages.cy.js:161]: @clari182
